src/mon/OSDMonitor.cc

```diff
@@ -970,31 +970,192 @@ void OSDMonitor::encode_pending(MonitorDBStore::TransactionRef t)
     tmp.apply_incremental(pending_inc);
 
     if (tmp.require_osd_release >= CEPH_RELEASE_LUMINOUS) {
```

Comment:
We'll want to rebase this as soon as https://github.com/ceph/ceph/pull/17322 merges, which removes the conditional